File: src/main/java/com/server/ordering/domain/Waiting.java
package com.server.ordering.domain;

import com.server.ordering.domain.member.Customer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;

import java.time.OffsetDateTime;
import java.time.ZoneId;
import java.time.ZonedDateTime;

import static javax.persistence.FetchType.*;
import static javax.persistence.GenerationType.*;
import static lombok.AccessLevel.PROTECTED;

@Entity
@Getter
@RequiredArgsConstructor
@NoArgsConstructor(access = PROTECTED)
public class Waiting {

    @Id @GeneratedValue(strategy = IDENTITY)
    @Column(name = "waiting_id")
    private Long id;

    @NonNull
    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "restaurant_id")
    private Restaurant restaurant;

    @NonNull
    @OneToOne(fetch = LAZY)
    @JoinColumn(name = "customer_id")
    private Customer customer;

    @NonNull
    private Integer myWaitingNumber;
    @NonNull
    private Byte numOfTeamMembers;

    @NonNull
    @OneToOne(fetch = LAZY)
    @JoinColumn(name = "phone_number")
    private PhoneNumber phoneNumber;

    private ZonedDateTime waitingRegisterTime;

    public void registerWaitingTime() {
        this.waitingRegisterTime = ZonedDateTime.now();
    }
}

File: src/main/kotlin/com/byeoru/ordering_server/domain/Waiting.kt
package com.byeoru.ordering_server.domain

import com.byeoru.ordering_server.domain.member.Customer
import java.time.ZonedDateTime
import javax.persistence.*

@Entity
class Waiting(restaurant: Restaurant,
              customer: Customer,
              myWaitingNumber: Int,
              numOfTeamMembers: Byte,
              phoneNumber: PhoneNumber) {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "waiting_id")
     var id: Long? = null
         protected set

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "restaurant_id")
     var restaurant: Restaurant = restaurant
         protected set

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
     var customer: Customer = customer
         protected set

     var myWaitingNumber: Int = myWaitingNumber
         protected set

     var numOfTeamMembers: Byte = numOfTeamMembers
         protected set

    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "phone_number")
     var phoneNumber: PhoneNumber = phoneNumber
         protected set

     var waitingRegisterTime: ZonedDateTime? = null
         protected set

    fun registerWaitingTime() {
        waitingRegisterTime = ZonedDateTime.now()
    }
}